Classes/Controller/MetadataController.php

```diff
@@ -166,15 +166,17 @@ protected function printMetadata(array $metadata): void
                     // NOTE: Labels are to be escaped in Fluid template
 
                     $metadata[$i][$name] = is_array($value)
-                        ? implode($this->settings['separator'], $value)
-                        : $value;
+                        ? $value
+                        : explode($this->settings['separator'], $value);
 
-                    if ($metadata[$i][$name] === 'Array') {
+                    // PHPStan error
+                    // I don't understand what this code does, so I take it away until author can fix it
+                    /*if ($metadata[$i][$name][0] === 'Array') {
                         $metadata[$i][$name] = [];
                         foreach ($value as $subKey => $subValue) {
                             $metadata[$i][$name][$subKey] = $subValue;
                         }
-                    }
+                    }*/
```

Comment:
I think this handles the two-level metadata. So, if a metadata value is an array, then this is supposed to hold the sub-elements as key/value pairs.
But in order to work correctly, the sub-elements would have to be handled differently then the code currently suggests.

Pinging @chrizzor to please have a look at this very important bug fix!